### PR TITLE
setup: Block non-alphabetical characters

### PR DESCRIPTION
**Pull request by mssola**
I've also fixed a redirect.

See bsc#1031475

Signed-off-by: Miquel Sabaté Solà <msabate@suse.com>

### CODE REVIEW

File: app/validators/alphabetic_validator.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    record.errors[attribute] << "contains invalid characters"
+  end
+
+  protected
+
+  # Returns true if the given value belongs to a protected key.
+  def protected?(pillar)
+    keys = Pillar.all_pillars
+    PROTECTED_VALUES.any? { |v| keys[v] == pillar }
```
Comment (ereslibre) [inline app/validators/alphabetic_validator.rb]: Can this be simplified as `PROTECTED_VALUES.include? pillar` ?

File: app/models/pillar.rb [hunk 2]
```diff
@@ -54,5 +54,22 @@ def all_pillars
...
+    # encountered errors.
+    def apply(pillars)
+      errors = []
+
+      Pillar.all_pillars.each do |key, pillar_key|
+        pillar = Pillar.find_or_initialize_by pillar: pillar_key
+        pillar.value = pillars[key]
+        next if pillar.save
+
+        exp = pillar.errors.messages.empty? ? "" : ": #{pillar.errors.messages[:value].first}"
```
Comment (ereslibre) [inline app/models/pillar.rb]: The test can be done with `pillar.errors.empty?`

Comment (ereslibre) [inline app/models/pillar.rb]: In the near future we can even add `pillar.errors.add(pillars[key], ...` to add errors to their specific virtual pillar attribute, and we could highlight specific fields on the forms, if required.

File: app/validators/alphabetic_validator.rb (hunk 1 above)
Comment (mssola) [inline app/validators/alphabetic_validator.rb]: Not really. That's what I tried at first, but bear in mind that the `PROTECTED_VALUES` constant only contain the key for which we identify the different pillar keys, which are the ones to be compared then after. A bit confusing, but we cannot simplify it :P

**Pull request merged**